components/job-orchestration/job_orchestration/search_scheduler/search_scheduler.py

```diff
@@ -0,0 +1,306 @@
+    db_cursor.execute(f"""
+        SELECT {SEARCH_JOBS_TABLE_NAME}.id as job_id,
+        {SEARCH_JOBS_TABLE_NAME}.status as job_status,
+        {SEARCH_JOBS_TABLE_NAME}.search_config,
+        {SEARCH_JOBS_TABLE_NAME}.submission_time
+        FROM {SEARCH_JOBS_TABLE_NAME}
+        WHERE {SEARCH_JOBS_TABLE_NAME}.status={JobStatus.PENDING}
+    """)
+    return db_cursor.fetchall()
+
+
+def fetch_cancelling_search_jobs(db_cursor) -> list:
+    db_cursor.execute(f"""
+        SELECT {SEARCH_JOBS_TABLE_NAME}.id as job_id
+        FROM {SEARCH_JOBS_TABLE_NAME}
+        WHERE {SEARCH_JOBS_TABLE_NAME}.status={JobStatus.CANCELLING}
+    """)
+    return db_cursor.fetchall()
+
+
+def set_job_status(
+    db_conn, job_id: str, status: JobStatus, prev_status: Optional[JobStatus] = None, **kwargs
+) -> bool:
+    cursor = db_conn.cursor()
+
+    field_set_expressions = [f'{k}="{v}"' for k, v in kwargs.items()]
+    field_set_expressions.append(f"status={status}")
+    update = f'UPDATE {SEARCH_JOBS_TABLE_NAME} SET {", ".join(field_set_expressions)} WHERE id={job_id}'
+
+    if prev_status is not None:
+        update += f' AND status={prev_status}'
+    cursor.execute(update)
+
+    db_conn.commit()
+
+    rval = cursor.rowcount != 0
+    cursor.close()
+    return rval
+
+
+def poll_and_handle_cancelling_search_jobs(db_conn) -> None:
+    global active_jobs
+
+    cursor = db_conn.cursor()
+    cancelling_jobs = fetch_cancelling_search_jobs(cursor)
+    db_conn.commit()
+    cursor.close()
+
+    for job_id, in cancelling_jobs:
+        if job_id in active_jobs:
+            cancel_job(job_id)
+        logger.info(f"Cancelled job {job_id}.")
+        set_job_status(db_conn, job_id, JobStatus.CANCELLED, prev_status=JobStatus.CANCELLING)
+
+
+def get_archives_for_search(
+    db_conn,
+    search_config: SearchConfig,
+):
+    query = f"""SELECT id as archive_id
+            FROM {CLP_METADATA_TABLE_PREFIX}archives
+            """
+    if search_config.end_timestamp is not None or search_config.begin_timestamp is not None:
+        query += " WHERE "
+    
+    need_and = False
+    if search_config.end_timestamp is not None:
+        query += f"begin_timestamp <= {search_config.end_timestamp}"
+        need_and = True
+
+    if search_config.begin_timestamp is not None:
+        if need_and:
+            query += " AND "
+        query += f"end_timestamp >= {search_config.begin_timestamp}"
+        need_and = True
+    
+    query += " ORDER BY end_timestamp DESC"
+
+    cursor = db_conn.cursor()
+    cursor.execute(
+        query
+    )
+    archives_for_search = [archive_id for archive_id, in cursor.fetchall()]
+    db_conn.commit()
+    cursor.close()
+    return archives_for_search
+
+
+def get_search_tasks_for_job(
+    archives_for_search: List[tuple],
+    job_id: str,
+    search_config: SearchConfig,
+    results_cache_uri: str,
+):
+    search_config_obj = search_config.dict()
+    return group(
+        search.s(
```

Comment:
Can we set the task ID to a monotonically increasing number so that it doesn't use a UUID and pollute our logs?

Reply:
It would also potentially give us some extra things to track across restarts depending on how Celery handles reused task IDs.

Reply:
Sure, we can leave it as UUIDs for now (but I'd argue we shouldn't rely on UUIDs to save us in a distributed setting, lol).